packages/cli/commands/worker.ts

```diff
@@ -197,6 +196,20 @@ export class Worker extends Command {
 			{ retryOf: currentExecutionDb.retryOf as string },
 		);
 
+		try {
+			await checkPermissionsForExecution(workflow, workflowOwner.id);
+		} catch (error) {
+			const failedExecution = generateFailedExecutionFromError(
+				currentExecutionDb.mode,
+				error,
+				error.node,
+			);
+			await additionalData.hooks.executeHookFunctions('workflowExecuteAfter', [failedExecution]);
+			return {
+				success: true,
```

Comment:
`success: true` here doesn't actually mean that the job ran successfully, but that the job should not be put back in the queue, right?

Reply:
Exactly, it means that the main main process (or webhook; whoever started this) can check the information from DB and finish whatever processes it needs to.